localstack/aws/accounts.py

```diff
@@ -57,10 +53,6 @@ def get_account_id_from_access_key_id(access_key_id: str) -> Optional[str]:
     if re.match(r"\d{12}", access_key_id):
         return access_key_id
 
-    # For internal calls, account ID is determined by the resource being accessed
-    elif access_key_id == INTERNAL_AWS_ACCESS_KEY_ID:
-        return None
-
```

Comment:
We might want to reconsider this. The idea was to avoid a fallback to a default account ID and fail instead with NoneType. The fallbacks have caused bugs that have been tricky to debug in the past.

Reply:
As discussed, all requests should contain some kind of account id, even for service principals (potentially), so it seems safe to assume that for now.